File: service-core/internal/modules/staff/domain/staff.go
package domain

import (
	"time"

	"github.com/google/uuid"
)

type Staff struct {
	ID     uuid.UUID
	UserID uuid.UUID

	CreatedAt time.Time
	UpdatedAt *time.Time
	DeletedAt *time.Time
}

// -------

// id
// account_id

// staff_type
// (name: personal/business)

// display_name
// description

// email
// phone

// address
// city
// province
// postal_code
// latitude
// longitude

// logo_url
// banner_url

// is_active
// is_verified

// bank_name
// bank_account_name
// bank_account_number

// created_at
// updated_at

File: control-panel/src/models/Staff.ts
export interface Staff {
  id: string;
  user_id: string;
  name: string;
  username: string;
  phone?: string | null;
  avatar_url?: string | null;
  created_at: string;
  updated_at?: string | null;
  last_login_at?: string | null;
}

export interface StaffListResponse {
  page: number;
  limit: number;
  total: number;
  staff: Staff[];
}

export interface AddStaffAccountPayload {
  email: string;
  name: string;
  username: string;
  password?: string;
  phone?: string;
}

export interface CreateStaffPayload {
  name: string;
  description?: string;
  logo_url?: string;
  banner_url?: string;
}
